File: python/drop_all_series.py
import requests

serverPath = "http://172.0.0.1:8086"
databaseName = "mydb"
user = "root"
password = "root"

url = serverPath + "/db/" + databaseName + "/series?"

r = requests.get (url, params = {'u':user, 'p':password, 'q': 'list series'})
data = r.json()
#print data
for var in data: 
	print "Dropping " + var["name"]; requests.get (url, params = {'u':user, 'p':password, 'q': 'drop series' + var["name"]});

File: haskell/drop_all_series.hs
{-# LANGUAGE OverloadedStrings #-}
module Main where 

import Network.HTTP
import Data.Aeson.Lens
import Control.Lens
import Data.Text

serverPath = "http://172.0.0.1"
databaseName = "mydb"
u = ("u", "root")
p = ("p", "root")

request :: [(String, String)] -> IO String
request urlvars = simpleHTTP (getRequest (serverPath ++ ":8086/db/" ++ databaseName ++ "/series?" ++ urlEncodeVars (u : p : urlvars))) >>= getResponseBody

requestDelete :: String -> IO String
requestDelete name = putStrLn ("Dropping " ++ name) >> request [("q", "drop series" ++ name)]

extractNames :: (Monad m, AsValue s) => s -> m [Text]
extractNames value = return (value ^.. values . key "name" . _String)

main :: IO ()
main = request [("q", "list series")] >>= extractNames >>= mapM_ (requestDelete . unpack)
